### PR TITLE
WIP - Root emails

### PR DESCRIPTION
**Pull request by yawn**
WIP for #13. Currently in it:

- Delegation zone with MX, DKIM and verification records
- Mail bucket and appropriate policies
- Website redirect configuration on the bucket to serve as zone delegation status health check (alarm missing)
- SES rule set creation for inbound mail handling and rule set activation

### CODE REVIEW

**Comment by yawn** [comment]
I'll rebase to just 4ab6532 after the review.

File: components/xx-rootmail.yaml [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+      WebsiteConfiguration:
+        RedirectAllRequestsTo:
+          HostName: "example.com"
+          Protocol: "https"
+
+  DomainBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref DomainBucket
+      PolicyDocument: !Sub |
```
Comment (s0enke) [inline components/xx-rootmail.yaml:49]: Nitpick: can be YAML as well, see e.g. https://github.com/superwerker/superwerker/blob/4d26141e3a4f325f0f5d742286bf560c14bde42c/components/00-control-tower.yaml#L151-L163

File: components/xx-rootmail.yaml [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+      HealthCheckTags:
+        - Key: Name
+          Value: RootEmailsHostedZoneDelegationHealthCheck
+
+  HostedZoneDelegationHealthCheckRecord:
+    Type: AWS::Route53::RecordSet
+    Properties:
+      AliasTarget:
+        DNSName: !Sub
+          - "${DomainBucket}.${Endpoint}"
```
Comment (s0enke) [inline components/xx-rootmail.yaml:97]: nitpick: quotes usually unnecessary

File: components/xx-rootmail.yaml [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+
+            print('RequestType: {}'.format(RequestType))
+            print('PhysicalResourceId: {}'.format(PhysicalResourceId))
+            print('LogicalResourceId: {}'.format(LogicalResourceId))
+
+            id = PhysicalResourceId
+
+            data = {}
+
+            if RequestType == CREATE:
```
Comment (s0enke) [inline components/xx-rootmail.yaml:166]: how about DELETE actions?

**Review by s0enke — APPROVED** [review]

**Comment by s0enke** [comment]
@yawn feel free to merge and continue with mainline development :)

File: components/xx-rootmail.yaml (hunk 3 above)
Comment (yawn) [reply components/xx-rootmail.yaml:166]: There is nothing created here, just outputs populated.

File: components/xx-rootmail.yaml (hunk 1 above)
Comment (yawn) [reply components/xx-rootmail.yaml:49]: I usually prefer JSON for policies - easier to generated / check with standard tools

File: components/xx-rootmail.yaml (hunk 2 above)
Comment (yawn) [reply components/xx-rootmail.yaml]: Fixed

**Comment by s0enke** [comment]
merging and then mainline dev

**Pull request merged**